File: Programmers/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;

class Solution {
    
    
    public int[] solution(String today, String[] terms, String[] privacies) {
        int todayValue = toDays(today);
        Map<String, Integer> termMap = new HashMap<>();
        List<Integer> result = new ArrayList<>();

        for (String term : terms) {
            String[] parts = term.split(" ");
            
            termMap.put(parts[0], Integer.parseInt(parts[1]) * 28);
        }

        for (int i = 0; i < privacies.length; i++) {
            String[] parts = privacies[i].split(" ");
            
            int collected = toDays(parts[0]);
            String type = parts[1];

            int expireDate = collected + termMap.get(type) - 1;
            
            if (expireDate < todayValue) {
                result.add(i + 1);
            }
        }

        return result.stream().mapToInt(Integer::intValue).toArray();
    }
    
    private int toDays(String date) {
        String[] d = date.split("\\.");
        
        int year = Integer.parseInt(d[0]);
        int month = Integer.parseInt(d[1]);
        int day = Integer.parseInt(d[2]);
        
        return year * 12 * 28 + month * 28 + day;
    }    
}

File: Programmers/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <sstream>

using namespace std;

int toDays(const string& date) {
    int y, m, d;
    char dot;
    stringstream ss(date);
    ss >> y >> dot >> m >> dot >> d;
    return y * 12 * 28 + m * 28 + d;
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    int todayValue = toDays(today);
    
    unordered_map<string, int> termMap;
    vector<int> result;

    for (const string& term : terms)
    {
        string name;
        int months;
        stringstream ss(term);
        
        ss >> name >> months;
        
        termMap[name] = months * 28;
    }

    for (int i = 0; i < privacies.size(); i++)
    {
        string date, type;
        stringstream ss(privacies[i]);
        
        ss >> date >> type;

        int collected = toDays(date);
        int expire = collected + termMap[type] - 1;

        if (expire < todayValue)
        {
            result.push_back(i + 1);
        }
    }

    return result;
}